Update /Core/Specification description (#1142)
spdx/spdx-3-model: model/Core/Classes/Specification.md

## Summary

> A detailed document that outlines the exact requirements for a product,

Reviewer: ```suggestion
A specification is a detailed description of the design, requirements, or features of a product, process, or system.
```  
It will not just be requirements but it can be features,design...

process, or system.

## Description

> A specification (spec) is a detailed document that outlines the exact

Reviewer: ```suggestion
A specification (spec) is a detailed document that outlines the 
```

> requirements for a product, process, or system.

Reviewer: ```suggestion
of the design, requirements, or features for a product, process, or system.
```

Author: Together with another suggestion to line 12, it will not be readable.
But I will just take all the suggestions first and will revise after.

> Specifications are often based on standards but provide more specific details

Reviewer: this might not be required.

Author: Thank you. I will remove it then.

Author: Removed
